src/intervals/construction.jl

```diff
@@ -0,0 +1,216 @@
+# defined after all methods they use.
+@generated function Interval{T}(x::Irrational) where T
+    res = Interval{T}(x(), x())  # Precompute the interval
+    return :(return $res)  # Set body of the function to return the precomputed result
+end
+
+Interval(x::Irrational) = Interval{default_bound()}(x)
+
+"""
+    interval(a, b)
+
+Create an interval, checking whether [a, b] is a valid `Interval`
+If so, then an `Interval(a, b)` object is returned;
+if not, a warning is printed and the empty interval is returned.
+"""
+function interval(a::T, b::S) where {T<:Real, S<:Real}
+    if !is_valid_interval(a, b)
+        @warn "Invalid input, empty interval is returned"
+        return emptyinterval(promote_type(T, S))
+    end
+
+    return Interval(a, b)
+end
+
+interval(a::Real) = interval(a, a)
+
+const checked_interval = interval
+
+"""
+    a..b
+    ..(a, b)
+
+Create the interval `[a, b]`.
+
+Validity of the interval is checked, but nothing is done to compensate for
+the fact that floating point literals are rounded to nearest when parsed.
+
+Use the string macro `I"[a, b]"` to ensure tight enclosure around the number
+that is typed in, even when it is not exactly representable as a floating point
+number (like `0.1`).
+"""
+..(a, b) = checked_interval(a, b)
+
+"""
+    a ± b
+
+Create the interval `[a - b, a + b]`.
+
+Despite using the center-radius notation for its creation, the interval is
+still represented by its bounds internally.
+"""
+a ± b = checked_interval(-(a, b, RoundDown), +(a, b, RoundUp))
+±(a::Interval, b) = Interval(-(a.lo, b, RoundDown), +(a.hi, b, RoundUp))
+
+"""
+    atomic(::Type{<:Interval}, x)
+
+Construct the tightest interval of a given type that contains the value `x`.
+
+If `x` is an `AbstractString`, the interval is created by calling `parse`.
+
+If `x` is an `AbstractFloat`, the interval is widen to two eps to be sure
+to contain the number that was typed in.
+
+Otherwise it is the same as using the `Interval` constructor directly.
+"""
+atomic(::Type{F}, x) where {F<:Interval} = F(x)
+atomic(::Type{F}, x::AbstractString) where {F<:Interval} = parse(F, x)
+
+function atomic(::Type{F}, x::AbstractFloat) where {T, F<:Interval{T}}
+    lo = T(x, RoundDown)
+    hi = T(x, RoundUp)
+    if x == lo
+        lo = prevfloat(lo)
+    end
+    if x == hi
+        hi = nextfloat(hi)
+    end
+    return Interval(lo, hi)
+end
+
+"""
+    big53(x::Interval{Float64})
+
+Create an equivalent `BigFloat` interval to a given `Float64` interval.
+"""
+function big53(a::Interval{Float64})
+    return setprecision(BigFloat, 53) do  # precision of Float64
+        return Interval{BigFloat}(a)
+    end
+end
+
+"""
+    big53(x::Float64)
+
+Convert `x` to `BigFloat`.
+"""
+function big53(x::Float64)
+    return setprecision(BigFloat, 53) do
+        return BigFloat(x)
+    end
+end
+
+float(x::Interval{T}) where T = atomic(Interval{float(T)}, x)
+big(x::Interval) = atomic(Interval{BigFloat}, x)
```

Comment:
I suggest to add docstrings to these functions. Notice that `float(a::Interval{BigFloat})` returns a bigfloat interval.